### DISCUSSION TITLE
Yarn 4 and Corepack not working

### DISCUSSION THREAD

**Discussion by fabri86** [vercel/community#6515] [Bug Report]
### Page to Investigate

https://vercel.com/fabri86s-projects/isswro21/EXY8jFtjGGE8VdeJccArH6obcvKB

### Steps to Reproduce

Hello,
I thought I was the last person on the earth running a project on yarn 1.22.19... 
So today I have finally decided to upgrade my project to use yarn 4, not even checking if Vercel supports yarn 2+ because it sounded obvious. I would like to remark that my project is not a SSR one nor a Serveless Function. It is a normal REACT app built with React 18 and Typescript.

I followed the documentation to enable Corepack by using an environment variable, and I did some troubleshooting after consulting the entire documentation/the currently open issues and discussion. Nothing helped.

![Screenshot 2024-04-15 at 13 46 14](https://github.com/vercel/community/assets/17066478/b050f846-2941-4745-9034-9183ac069167)

By reading around, it seems I am not the only user affected:
https://stackoverflow.com/questions/78202142/vercel-this-projects-package-json-defines-packagemanager-yarn4-1-1-howe

Is there 
- any planned fix for this problem or... 
- any workaround to start using yarn 4 on Vercel or..
- any work planned to support yarn 4?

If Yarn 4 is the future and Vercel is the future, it would be cool if we could have only one future, all together.
Please let me and many other of your users know something.

Thanks in advance!

**Comment by akdombrowski** [comment]
I found an article stating yarn 2 is only supported for static builds. Since I couldn't find anything else on yarn 3+ being supported either, this is probably it.
https://vercel.com/guides/does-vercel-support-yarn-2

Someone else did open a feature request. I suggest upvoting and/or commenting there to raise awareness!
https://github.com/orgs/vercel/discussions/5664#discussion-6149981
#5664

[a comment by an automated account is not shown]
